src/library_wasmfs_fetch.js

```diff
@@ -67,12 +74,14 @@ mergeInto(LibraryManager.library, {
 
       // read/getSize fetch the data, then forward to the parent class.
       read: async (file, buffer, length, offset) => {
-        await getFile(file);
-        return jsFileOps.read(file, buffer, length, offset);
+        return getFile(file)
+        .then(() => jsFileOps.read(file, buffer, length, offset))
+        .catch((status) => status === 404 ? -{{{ cDefine('ENOENT') }}} : -{{{ cDefine('EBADF') }}});
       },
       getSize: async(file) => {
-        await getFile(file);
-        return jsFileOps.getSize(file);
+        return getFile(file)
+        .then(() => jsFileOps.getSize(file))
+        .catch((status) => jsFileOps.getSize(file));
```

Comment:
How about this? (since we call the same code in the end anyhow)
```suggestion
        try {
          await getFile(file);
        } catch(e) {}
        return jsFileOps.getSize(file);
```

Reply:
Yes, it looks much better, thanks.